src/csvwidget/widget.ts

```diff
@@ -217,6 +217,27 @@ function createDelimiterSwitcherNode(): HTMLElement {
 export
 class CSVWidgetFactory extends ABCWidgetFactory<CSVWidget, DocumentRegistry.IModel> {
   /**
+   * The name of the widget to display in dialogs.
+   */
+  get name(): string {
+    return 'Table';
+  }
+
+  /**
+   * The file extensions the widget can view.
+   */
+  get fileExtensions(): string[] {
+    return ['.csv'];
```

Comment:
When we originally went back and forth on making the name and file extension behavior part of the registration, we decided on the registration so that it would be very easy to change the file extension registration and displayed names without having to subclass and write code - you just change a bit of metadata at registration. Why put it back in the factory now?